### PR TITLE
Move ResetServiceIT to Sandbox integration tests

### LINKED ISSUE #2857: Port ResetServiceIT to the Ledger API Test Tool
Port `ResetServiceIT` to the Ledger API Test Tool.

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Closes #2857

- Also deals with the long ignored todo about not using the context
- Attempts at restoring this as a non-flaky test
- Attempts at running the same test across runtime environments

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+final class ResetServiceIT
+    extends AsyncWordSpec
+    with AsyncTimeLimitedTests
+    with Matchers
+    with ScalaFutures
+    with SandboxFixture
+    with SuiteResourceManagementAroundEach
+    with TestCommands {
+
+  private val eventually = RetryStrategy.exponentialBackoff(10, 10.millis)
```
Comment (SamirTalwar) [inline ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala]: Why not use ScalaTest's `eventually`?

File: ledger/test-common/src/main/scala/com.digitalasset.platform.testing/WaitForCompletionsObserver.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import com.digitalasset.ledger.api.v1.command_completion_service.CompletionStreamResponse
+import io.grpc.Context
+import io.grpc.stub.StreamObserver
+
+import scala.concurrent.{Future, Promise}
+
+object WaitForCompletionsObserver {
```
Comment (SamirTalwar) [inline ledger/test-common/src/main/scala/com.digitalasset.platform.testing/WaitForCompletionsObserver.scala]: This seems to be basically the same as `SizeBoundObserver`, with the only real difference being that this one fails when `onCompleted` is called, and `SizeBoundObserver` succeeds. Can you make that one do what you want instead?

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        } yield {
+          lid1 should not equal lid2
+          IsStatusException(Status.Code.NOT_FOUND)(throwable)
+        }
+      }
+
+      "return new ledger ID - 20 resets" in {
+        Future
+          .sequence(Iterator.iterate(getLedgerId())(_.flatMap(reset)).take(20).toVector)
+          .map(ids => ids.distinct should have size 20L)
```
Comment (SamirTalwar) [inline ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala]: I think I'd prefer it if we could use `for` comprehensions with assertions.

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+              .iterate(getLedgerId()) { ledgerIdF =>
+                for {
+                  ledgerId <- ledgerIdF
+                  _ <- submitAndExpectCompletions(ledgerId, numberOfCommands)
+                  (newLedgerId, timing) <- timedReset(ledgerId) if timing <= 5.seconds
+                } yield newLedgerId
+              }
+              .take(numberOfAttempts)
+          )
+          .map(_ => succeed)
```
Comment (SamirTalwar) [inline ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala]: Even when the assertion is just a `succeed`, I think I'd still prefer a `for`, but I'm less fussed about this one.

File: ledger/test-common/src/main/scala/com.digitalasset.platform.testing/WaitForCompletionsObserver.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/test-common/src/main/scala/com.digitalasset.platform.testing/WaitForCompletionsObserver.scala]: Not really, this is specialized to count the number of completions within each response, not to count each response as one.

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala (hunk 4 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala]: There are different opinions in the team (@gerolf-da usually prefers `map` when the call is very simple). If you don't have a strong argument against this one I'd keep it as it is right now, mostly out of sheer inertia. Ideally I would have wanted to use `all(measurements) should be <= threshold` but for some reason it doesn't compile.

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala]: I'm not sure of the reason why but everything fails if I use that. I'm not sure of the semantics there.

**Pull request merged**